store/v2/storage/sqlite/db.go

```diff
@@ -210,12 +210,19 @@ func (db *Database) Prune(version uint64) error {
 	}
 
 	// prune removed stores
-	pruneRemovedStoreKeysStmt := `DELETE FROM state_storage
-	WHERE store_key IN (
-		SELECT key FROM state_storage WHERE store_key = ? AND value = ? AND version <= ?
+	pruneRemovedStoreKeysStmt := `DELETE FROM state_storage AS s
+	WHERE EXISTS ( 
+		SELECT * FROM
+			(
+			SELECT key, MAX(version) AS max_version
+			FROM state_storage
+			WHERE store_key = ? AND value = ? AND version <= ?
+			GROUP BY key
+			) AS t
+		WHERE s.store_key = t.key AND s.version <= t.max_version
```

Comment:
I have tested this locally. The `EXISTS` operator feels a bit strange but it works with the where clause  👍